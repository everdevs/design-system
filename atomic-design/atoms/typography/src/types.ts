import { PropsWithTheme } from "@evernest/theme";

export type TypographyElement = HTMLDivElement;

export enum TypographyVariant {
	h1 = "h1",
	h2 = "h2",
	h3 = "h3",
	body = "body",
}

export interface RawStyledTypographyProps extends PropsWithTheme<TypographyElement> {
	/** Add types */
}

export interface StyledTypographyProps extends RawStyledTypographyProps {
	/** Add types */
	/** @default TypographyVariant.body */
	variant?: TypographyVariant;
	component?: keyof JSX.IntrinsicElements;
}
