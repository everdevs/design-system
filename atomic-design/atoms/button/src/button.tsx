import { css } from "@emotion/core";
import styled from "@emotion/styled";
import { ButtonProps, Size, Variation } from "./types";

const Button = styled("button", {
	shouldForwardProp: (propName: string) => !["theme"].includes(propName),
})<ButtonProps>`
	cursor: pointer;
	border: 0;
	${({ theme, size, variation }) => css`
		box-shadow: ${theme.tokens.shadow.medium.css};
		height: ${size === Size.small ? "20px" : "40px"};
	`};
`;

Button.defaultProps = {
	type: "button",
	size: Size.medium,
	variation: Variation.default,
};

export { Button };
