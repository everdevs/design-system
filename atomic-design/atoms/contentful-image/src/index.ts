export * from "./contentful-image";
export * from "./utils";
export * from "./types";
