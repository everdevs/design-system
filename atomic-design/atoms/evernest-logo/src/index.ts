export * from "./evernest-logo";
export * from "./types";
