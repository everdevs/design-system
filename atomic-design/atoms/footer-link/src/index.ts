export * from "./footer-link";
export * from "./types";
