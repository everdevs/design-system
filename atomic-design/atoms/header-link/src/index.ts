export * from "./header-link";
export * from "./types";
