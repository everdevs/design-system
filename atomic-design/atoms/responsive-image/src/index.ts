export * from "./responsive-image";
export * from "./utils";
export * from "./types";
