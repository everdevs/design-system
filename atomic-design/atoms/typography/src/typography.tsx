import styled from "@emotion/styled";
import React from "react";
import { StyledTypographyProps, TypographyElement, TypographyVariant } from "./types";

export const StyledHeadline1 = styled.h1<StyledTypographyProps>`
	font-family: var(--font-face-serif);
	font-weight: normal;
	font-size: 55px;
	line-height: 63px;
`;
export const StyledHeadline2 = styled.h2<StyledTypographyProps>`
	font-family: var(--font-face-serif);
	font-weight: normal;
	font-size: 36px;
	line-height: 45px;
`;
export const StyledHeadline3 = styled.h3<StyledTypographyProps>`
	font-family: var(--font-face-serif);
	font-weight: normal;
	font-size: 24px;
	line-height: 36px;
`;

export const StyledHeadline4 = styled.h4<StyledTypographyProps>`
	font-family: var(--font-face-serif);
	font-weight: normal;
	font-size: 20px;
	line-height: 24px;
	margin: 0;
`;

export const Body = styled.div<StyledTypographyProps>`
	font-family: var(--font-face-sans-serif);
	font-weight: lighter;
	font-size: 16px;
	line-height: 28px;
`;

export const Body2 = styled.div<StyledTypographyProps>`
	font-family: var(--font-face-sans-serif);
	font-weight: normal;
	font-size: 20px;
	line-height: 30px;
`;

const variants = {
	[TypographyVariant.h1]: StyledHeadline1,
	[TypographyVariant.h2]: StyledHeadline2,
	[TypographyVariant.h3]: StyledHeadline3,
	[TypographyVariant.h4]: StyledHeadline4,
	[TypographyVariant.body]: Body,
	[TypographyVariant.body2]: Body2,
};

export const Typography = React.forwardRef<TypographyElement, StyledTypographyProps>(
	({ variant, component, ...props }, ref) => {
		const Variant = React.useMemo(() => (variant ? variants[variant] : Body), [variant]);
		const Component = React.useMemo(
			() => (component ? Variant.withComponent(component) : Variant),
			[Variant, component]
		);
		return <Component {...props} ref={ref} />;
	}
);

Typography.defaultProps = {
	variant: TypographyVariant.body,
};
