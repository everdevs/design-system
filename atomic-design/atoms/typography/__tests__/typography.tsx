import React from "react";
import { mount } from "enzyme";
import { Typography, TypographyVariant } from "../src";
import { Wrapper, createTree } from "@evernest/dev-helpers";

const text = "Hello World!";

test("Typography renders children", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.body}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find(Typography).text()).toEqual(text);
});

test("Typography allows setting classname", () => {
	const className = "test";
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.body} className={className} />
		</Wrapper>
	);
	expect(wrapper.find(Typography)).toHaveClassName(className);
});

test("Typography renders h1", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.h1}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find("h1").length).toEqual(1);
});

test("Typography renders h2", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.h2}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find("h2").length).toEqual(1);
});

test("Typography renders h3", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.h3}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find("h3").length).toEqual(1);
});

test("Typography renders h4", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.h4}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find("h4").length).toEqual(1);
});

test("Typography renders body as p", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.body}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find("p").length).toEqual(1);
});

test("Typography renders body2 as p", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.body2}>{text}</Typography>
		</Wrapper>
	);
	expect(wrapper.find("p").length).toEqual(1);
});

test("Typography renders h2 with component h1", () => {
	const wrapper = mount(
		<Wrapper>
			<Typography variant={TypographyVariant.h2} component="h1">
				{text}
			</Typography>
		</Wrapper>
	);
	expect(wrapper.find("h1").length).toEqual(1);
});
