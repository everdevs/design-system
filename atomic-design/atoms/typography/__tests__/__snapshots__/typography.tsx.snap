// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography matches snapshot 1`] = `
<div
  className="css-qmizr8"
>
  <h1
    className="css-1uyacqh"
  >
    Hello World!
  </h1>
  <h2
    className="css-1qroef3"
  >
    Hello World!
  </h2>
  <h3
    className="css-e26qad"
  >
    Hello World!
  </h3>
  <h4
    className="css-10bnnn3"
  >
    Hello World!
  </h4>
  <p
    className="css-9gqsre"
  >
    Hello World!
  </p>
  <p
    className="css-1tl4u9d"
  >
    Hello World!
  </p>
</div>
`;
